### PR TITLE
feat(persisted): Add support for persisted queries.

### PR DESCRIPTION
**Pull request by salmanm**
### What does this PR do?

- This PR adds a `persisted` option support to `useQuery`. It's useful to tell the backend graphql server if you're sending query SHA instead of the actual query text.

https://github.com/mcollina/fastify-gql/pull/139 adds in support for the same on server side.

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

File: packages/babel-plugin-extract-gql/examples/fastify-gql/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "scripts": {
+    "start": "node ."
+  },
+  "author": "Salman Mitha <SalmanMitha@gmail.com>",
+  "license": "MIT",
+  "dependencies": {
+    "fastify": "^2.13.1",
+    "fastify-cors": "^3.0.3",
+    "fastify-gql": "salmanm/fastify-gql"
```
Comment (mcollina) [inline packages/babel-plugin-extract-gql/examples/fastify-gql/package.json:14]: This needs to be updated, we merged the code in fastify-gql.

File: packages/babel-plugin-extract-gql/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "babel-plugin-extract-gql",
+  "version": "2.1.0",
```
Comment (mcollina) [inline packages/babel-plugin-extract-gql/package.json:3]: Is this correct? How does it work with lerna?

File: packages/babel-plugin-extract-gql/test/unit/fixtures/actual/.4.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useQuery, persist } from 'graphql-hooks'
```
Comment (mcollina) [inline packages/babel-plugin-extract-gql/test/unit/fixtures/actual/.4.js:1]: why is this file `.4.js` instead `4.js`?

File: .eslintignore [hunk 4]
```diff
@@ -3,3 +3,4 @@ dist
 es
 lib
 build
+**/fixtures
```
Comment (mcollina) [inline .eslintignore:6]: why are we ignoring them in eslint?

Comment (salmanm) [reply .eslintignore:6]: Coz the unit test files are partial component code and linter might complain for certain things. Also, `fixtures/expected` contains code that babel outputs, which is not linter compliant.

File: packages/babel-plugin-extract-gql/test/unit/fixtures/actual/.4.js (hunk 3 above)
Comment (salmanm) [reply packages/babel-plugin-extract-gql/test/unit/fixtures/actual/.4.js:1]: To disable this test case. Basically this should cover the case where an external options object is supplied to `useQuery` (e.g. `useQuery(quer, optObj)`). The code doesn't yet know how to manipulate that object.

But simpler cases like `useQuery(query)` OR `useQuery(query, { a: 1 })` are covered and working fine.

I was hoping to come back and fix the 4th test.

File: .eslintignore (hunk 4 above)
Comment (mcollina) [reply .eslintignore:6]: ok

**Pull request merged**